airbyte-webapp/src/components/connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -21,42 +21,49 @@ const clearBreakingFieldChanges = (nodeStream: SyncSchemaStream, breakingChanges
     return nodeStream;
   }
 
+  const { primaryKey, cursorField } = nodeStream.config;
+
   let clearPrimaryKey = false;
   let clearCursorField = false;
 
   for (const streamTransformation of breakingChangesByStream) {
-    // get all of the removed field paths for this transformation
-    const removedFieldPaths = streamTransformation.updateStream?.map((update) => update.fieldName);
-
-    if (!removedFieldPaths?.length) {
+    if (!streamTransformation.updateStream || !streamTransformation.updateStream?.length) {
       continue;
     }
 
+    // get all of the removed field paths for this transformation
+    const breakingFieldPaths = streamTransformation.updateStream
+      .filter(({ breaking }) => breaking)
```

Comment:
This is the fix. The rest of the code is cleanup.

Reply:
It looks like an empty `breakingFieldPaths` does not trigger a continue when it did before.
But below it wouldn't set either of the flags. Is this cleanup just removing an early exit of the loop?

Reply:
I did the cleanup before the fix, but I moved it up because the map is only necessary if the list has anything to change:

```
streamTransformation.updateStream?.length === streamTransformation.updateStream?.map((update) => update.fieldName).length
```